presto-main/src/main/java/io/prestosql/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -165,6 +178,78 @@ private void analyze(Expression expression)
         }
     }
 
+    private static Expression normalize(Expression expression)
+    {
+        return ExpressionTreeRewriter.rewriteWith(new ExpressionNormalizer(), expression);
+    }
+
+    private static class ExpressionNormalizer
+            extends ExpressionRewriter<Void>
+    {
+        @Override
+        public Expression rewriteCast(Cast node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+        {
+            return new Cast(
+                    node.getExpression(),
+                    normalizeType(node.getType()),
+                    node.isSafe(),
+                    node.isTypeOnly());
+        }
+
+        private DataType normalizeType(DataType type)
+        {
+            if (type instanceof IntervalDayTimeDataType) {
+                return type;
+            }
+
+            if (type instanceof DateTimeDataType) {
+                return type;
+            }
+
+            if (type instanceof GenericDataType) {
+                GenericDataType genericDataType = (GenericDataType) type;
+                return new GenericDataType(
+                        Optional.empty(),
+                        normalizeIdentifier(genericDataType.getName()),
+                        genericDataType.getArguments().stream()
+                                .map(this::normalizeTypeParameter)
+                                .collect(toImmutableList()));
+            }
+
+            if (type instanceof RowDataType) {
+                new RowDataType(
+                        Optional.empty(),
+                        ((RowDataType) type).getFields().stream()
+                                .map(field -> new RowDataType.Field(
+                                        Optional.empty(),
+                                        field.getName().map(this::normalizeIdentifier),
+                                        normalizeType(field.getType())))
+                                .collect(toImmutableList()));
+            }
+
+            return type;
+        }
+
+        private Identifier normalizeIdentifier(Identifier identifier)
+        {
+            if (identifier.isDelimited()) {
+                return identifier;
+            }
+            return new Identifier(identifier.getValue().toLowerCase(ENGLISH));
```

Comment:
FYI, standard SQL normalization rule is to use upper-case.

For example, `“FOO”` should match `foo`, but `“foo”` should not.